### PR TITLE
feat: Add rust code in binary_tree.md

### PR DESCRIPTION
**Pull request by lhxsm**
If this PR is related to coding or code translation, please fill out the checklist and paste the console outputs to the PR.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

File: docs/chapter_tree/binary_tree.md [hunk 1]
```diff
@@ -126,7 +126,12 @@
 === "Rust"
 
     ```rust title=""
-
+    /* 二叉树节点结构体 */
+    struct Node<T> {
+        value: T,
+        left: Option<Box<Node<T>>>,
```
Comment (night-cruise) [inline docs/chapter_tree/binary_tree.md:132]: The definition of tree node should be like this:
```rust
use std::cell::RefCell;
use std::rc::Rc;

/* 二叉树节点结构体 */
struct TreeNode {
    val: i32,
    left: Option<Rc<RefCell<TreeNode>>>,
    right: Option<Rc<RefCell<TreeNode>>>,
}
```
You can find it in the  [tree_node.rs](https://github.com/krahets/hello-algo/blob/main/codes/rust/include/tree_node.rs).

And an instantiation method needs to be added:
```rust
impl TreeNode {
    /* 节点初始化方法 */
    fn new(val: i32) -> Self {
        Self { val, left: None, right: None }
    }
}
```

**Comment by krahets** [comment]
Hi @lhxsm , please update the PR according to the comments.

**Comment by lhxsm** [comment]
> Hi @lhxsm , please update the PR according to the comments.

Sorry, should I be consistent with the commets or use the method I submitted?

**Comment by krahets** [comment]
> > Hi @lhxsm , please update the PR according to the comments.
> 
> Sorry, should I be consistent with the commets or use the method I submitted?

Update the PR according to the comments from @night-cruise

**Review by krahets — APPROVED** [review]

**Pull request merged**